app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsDetailsActions/PipelineDetailsActionsButton/index.js

```diff
@@ -98,6 +102,40 @@ export default class PipelineDetailsActionsButton extends Component {
     duplicatePipeline(this.props.pipelineName, sanitizeConfig(this.pipelineConfig));
   };
 
+  toggleDiscardConfirmation = () => {
+    this.setState({
+      showDiscardConfirmation: !this.state.showDiscardConfirmation,
+    });
+  };
+
+  discardAndStartNewEdit = () => {
+    this.toggleDiscardConfirmation();
+    editPipeline(this.props.pipelineName);
+  };
+
+  handlePipelineEdit = () => {
+    if (!this.props.editDraftId) {
+      this.editConfigAndNavigate(this.props.pipelineName);
```

Comment:
seems `editConfigAndNavigate` is only called in this func, then no need to have it on line139?

simple put `editPipeline(this.props.pipelineName, sanitizeConfig(this.pipelineConfig));`